v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/dbutils/dml/CassandraTypeHandler.java

```diff
@@ -0,0 +1,515 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.templates.dbutils.dml;
+import org.json.JSONObject;
+
+import java.nio.ByteBuffer;
+import java.sql.Timestamp;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.*;
+import java.util.stream.Collectors;
+
+class CassandraTypeHandler {
+
+    @FunctionalInterface
+    public interface TypeParser<T> {
+        T parse(Object value);
+    }
+
+    /**
+     * Generates a Type based on the provided {@link CassandraTypeHandler}.
+     *
+     * @param colName - which is used to fetch Key from valueJSON.
+     * @param valuesJson - contains all the key value for current incoming stream.
+     *
+     * @return a {@link Boolean} object containing the value represented in cassandra type.
+     */
+    public static Boolean handleCassandraBoolType(String colName, JSONObject valuesJson) {
+        return valuesJson.getBoolean(colName);
+    }
+
+    /**
+     * Generates a Type based on the provided {@link CassandraTypeHandler}.
+     *
+     * @param colName - which is used to fetch Key from valueJSON.
+     * @param valuesJson - contains all the key value for current incoming stream.
+     *
+     * @return a {@link Float} object containing the value represented in cassandra type.
+     */
+    public static Float handleCassandraFloatType(String colName, JSONObject valuesJson) {
+        return valuesJson.getBigDecimal(colName).floatValue();
+    }
+
+    /**
+     * Generates a Type based on the provided {@link CassandraTypeHandler}.
+     *
+     * @param colName - which is used to fetch Key from valueJSON.
+     * @param valuesJson - contains all the key value for current incoming stream.
+     *
+     * @return a {@link Double} object containing the value represented in cassandra type.
+     */
+    public static Double handleCassandraDoubleType(String colName, JSONObject valuesJson) {
+        return valuesJson.getBigDecimal(colName).doubleValue();
```

Comment:
In case the column value is null valuesJson.getBigDecimal() will throw an exception right ? Shouldn't we use  something like `optBigDecimal` here

Reply:
we have handle the null case fo all the case @shreyakhajanchi

Reply:
Hi @shreyakhajanchi Resolving this as we have addressed the comment, Thanks